ntbs-service/DataAccess/UserRepository.cs

```diff
@@ -34,6 +34,12 @@ public UserRepository(NtbsContext context, IOptionsMonitor<AdOptions> adOptions)
             _context = context;
         }
 
+        public UserRepository(NtbsContext context, AdOptions adOptions)
+        {
+            _adOptions = adOptions;
+            _context = context;
+        }

```

Comment:
As Josh has already pointed out, creating a new constructor just for unit tests is not ideal. However, making an `OptionsMonitor` used by the other constructor above is messy, using existing C# classes we have to set up a monitor factory, which is a faff, or I need to make a test of `IOptionsMonitor` (see [this SO post](https://stackoverflow.com/a/54140119)), which felt a little heavyweight. If you think making a test options monitor is worthwhile then I definitely can!

Reply:
You should be able to do this via a mock, which I think would be cleaner - i.e. in your tests have something like:
```
    var optionsMonitor = new Mock<IOptionsMonitor<AdOptions>>();
    optionsMonitor.Setup(om => om.CurrentValue).Returns(new AdOptions {ReadOnlyUserGroup = "ReadOnly"});
```

Reply:
Thanks for this! I had tried using a mock with no success (but I can't remember exactly what I did), but this works perfectly. Changed and the new `UserRepository` constructor removed